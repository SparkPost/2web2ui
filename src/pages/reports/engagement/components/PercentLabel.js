import React from 'react';

const LABEL = { height: 24, width: 64 };

// NOTE: ReferenceArea was used because they could easily attach to two bars and provide the
// coordinates and dimmenions to reliable place this label
export default function PercentLabel({
  percentage,
  width, // area around each bar, these touch and sum up to the total width
  x,
  yAxis
}) {
  const offset = width - (LABEL.width / 2);
  const y = yAxis.y + (yAxis.height / yAxis.tickCount) * Math.round(yAxis.tickCount / 2);

  return (
    <g>
      <svg {...LABEL} x={x + offset} y={y - LABEL.height / 2 + 1}>
        <path
          d="M3 1h52c.7 0 1.3.3 1.6.8l6.4 9c.5.7.5 1.7-.1 2.4l-6.3 8c-.4.5-1 .8-1.6.8H3a2 2 0 0 1-2-2V3c0-1.1.9-2 2-2z"
          fill="white"
          stroke="#e1e1e6"
          strokeMiterlimit="10"
          vectorEffect="non-scaling-stroke"
        />
      </svg>
      <text x={x + offset + 5} y={y + 6} fill="#55555a">{percentage}</text>
    </g>
  );
}
