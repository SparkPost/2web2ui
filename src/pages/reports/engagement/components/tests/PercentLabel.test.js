import React from 'react';
import { shallow } from 'enzyme';
import PercentLabel from '../PercentLabel';

describe('PercentLabel', () => {
  it('renders label', () => {
    const props = {
      percentage: '89.88%',
      width: 200,
      x: 64,
      yAxis: {
        height: 290,
        tickCount: 5,
        y: 25
      }
    };
    const wrapper = shallow(<PercentLabel {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
