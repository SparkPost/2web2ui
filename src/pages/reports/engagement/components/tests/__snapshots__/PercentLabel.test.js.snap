// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PercentLabel renders label 1`] = `
<g>
  <svg
    height={24}
    width={64}
    x={232}
    y={188}
  >
    <path
      d="M3 1h52c.7 0 1.3.3 1.6.8l6.4 9c.5.7.5 1.7-.1 2.4l-6.3 8c-.4.5-1 .8-1.6.8H3a2 2 0 0 1-2-2V3c0-1.1.9-2 2-2z"
      fill="white"
      stroke="#e1e1e6"
      strokeMiterlimit="10"
      vectorEffect="non-scaling-stroke"
    />
  </svg>
  <text
    fill="#55555a"
    x={237}
    y={205}
  >
    89.88%
  </text>
</g>
`;
